services/api/service.go

```diff
@@ -1335,9 +1333,47 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 		}
 	}()
 
+	// Get the current (in-memory) top bid value
+	api.builderTopBidValueLock.RLock()
+	localTopBidValue := api.builderTopBidValue[payload.ParentHash()]
```

Comment:
Is the parent hash the only thing that can differ for the payload to still be valid? Maybe a comment could be useful explaining why that's the case